### PR TITLE
Update DG Structure

### PR DESCRIPTION
**Pull request by simonteozw**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,35 +1,111 @@
...
+    3. [Logic Component](#logic)
+    4. [Model Component](#model)
+    5. [Common Classes](#common)
+4. [**Implementation**](#implementation)
+    1. [Module Manager](#module-manager)
+    2. [Lesson Manager](#lesson-manager)
+    3. [Task Manager](#task-manager)
+    4. [Contact Manager](#contact-manager)
+    5. [Upcoming Tab](#upcoming-tab)
+    6. [App-Wide Search](#app-wide-search)
```
Comment (UncleGrandpa925) [inline docs/DeveloperGuide.md]: we don't have the app-wide search anymore right? It will now be a contact tab?

**Review by UncleGrandpa925 — COMMENTED** [review]
Looks great! I have just one single question!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (simonteozw) [reply docs/DeveloperGuide.md]: Ok I shall remove it then. There is already a Contact Manager section

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1,35 +1,108 @@
...
+    2. [UI Component](#ui)
+    3. [Logic Component](#logic)
+    4. [Model Component](#model)
+    5. [Common Classes](#common)
+4. [**Implementation**](#implementation)
+    1. [Module Manager](#module-manager)
+    2. [Lesson Manager](#lesson-manager)
+    3. [Task Manager](#task-manager)
+    4. [Upcoming Tab](#upcoming-tab)
+    5. [App-Wide Search](#app-wide-search)
```
Comment (UncleGrandpa925) [inline docs/DeveloperGuide.md]: Heyy you still need to edit it here too!

**Review by UncleGrandpa925 — CHANGES_REQUESTED** [review]
Looks good! I just found one minor error 😄

**Review by UncleGrandpa925 — APPROVED** [review]

**Pull request merged**